src/importexport/capella/internal/capella.cpp

```diff
@@ -2138,7 +2138,7 @@ int Capella::readInt()
 
 char* Capella::readString()
 {
-    unsigned len = readUnsigned();
+    qint64 len = readUnsigned();
     char* buffer = new char[len + 1];
```

Comment:
Perhaps:
```suggestion
    unsigned len = readUnsigned();
    char* buffer = new char[static_cast<size_t>(len) + 1];
```
To make clear that the cast to size_t happens. To my knowledge, `operator new[]` takes size_t anyway, so with that change we're only making clear that the cast should happen before adding 1 rather than after.

Reply:
`qint64`, as taken by `read()` is a `long long` though. So we're a bit between a rock and a hard place, get the return valie for `readUnsigned()`and the arg for `new` right or the arg for `read()`

Reply:
But I don't really mine either way, just let me know what you prefer

Reply:
Perhaps my suggestion has the theoretical advantage that it contains only conversions that "fit" (i.e. a uint32 fits both in an int64 and in a size_t (≈uint64)). (In practice it will fit anyway though.) 
And I have a slight preference for using the correct return type for `readUnsigned`, for clarity.

Reply:
OK, done